types/serverless/plugins/aws/provider/awsProvider.d.ts

```diff
@@ -514,7 +515,7 @@ declare namespace Aws {
         tags?: Tags;
         vpc?: Vpc;
         package?: Package;
-        layers?: string[];
+        layers?: string[] | { [key: string]: string };
```

Comment:
This property needs to be an array of AWS ARN - see https://github.com/serverless/serverless/blob/master/lib/plugins/aws/provider/awsProvider.js#L957 refering to https://github.com/serverless/serverless/blob/b7901cdb77cb2c81dee62cb614d39d5d2fc824ff/lib/plugins/aws/provider/awsProvider.js#L396-L399 which can be a string or an object defining any cloudformation intrinsic function.
I'd go for the following solution if you agree
```suggestion
        layers?: (string|Record<string,string>)[]
```